### PR TITLE
Let tables be searched for by display name

### LINKED ISSUE #10665: Universal search can't find tables by Metabase friendly name
**Describe the bug**
Addition of tables to the universal search bar is great, but just found that it will only search on the actual table name, not the friendly name used everywhere else in Metabase. Note that these are the friendly names automatically generated by Meatbase, not manually overwritten.

**Logs**
No logs

**To Reproduce**
1. Have a table name in a database that maps to a friendly name with multiple words in Metabase e.g. "ServiceRequestType" in SQL Server -> "Service Request Type" in Metabase
2. In search bar type the friendly name as shown in the Admin > Data Model screen. e.g. "Service Request Type"
3. The table known everywhere in Metabase as "Service Request Type" is not listed
4. Search for the actual table name "ServiceRequestType" and the table is listed but not by it's friendly name.

**Expected behavior**
I would expect the search to be able to find tables based on how they are referenced in the rest of Metabase, i.e. by the friendly name.

**Screenshots**
Admin showing the table friendly name
image

Admin showing the actual table name
image

Searching by friendly name
image

Searching by actual table name
image

**Information about your Metabase Installation:**
- Your browser and the version: Chrome 76.0.3809.100
- Your operating system: Windows 7
- Your databases: SQL Server
- Metabase version: v0.33.0
- Metabase hosting environment: Jar file on Windows
- Metabase internal database: H2

**Severity**
Annoyance. Users can't search for tables by the names they know them by. This limits the usefulness of the improved search function.

**Additional context**

### PR DESCRIPTION
**Pull request by tsmacdonald**
[Fixes #10665]

Also lets us easily add other columns to search by if we want to…just not sure what those would be right now

### CODE REVIEW

File: test/metabase/api/search_test.clj [hunk 1]
```diff
@@ -327,26 +327,31 @@
...
+        (do-test-users [user [:crowberto :rasta]]
+          (is (= []
+                 (search-request user :q table-name)))))))
+  (testing "you should not be able to see a Table if the current user doesn't have permissions for that Table"
+    (mt/with-temp* [Database [{db-id :id}]
+                    Table    [table {:db_id db-id}]]
+      (perms/revoke-permissions! (group/all-users) db-id)
+      (is (= []
+             (binding [*search-request-results-database-id* db-id]
+               (search-request :rasta :q (:name table))))))))
```
Comment (tsmacdonald) [inline test/metabase/api/search_test.clj:354]: The later tests were all nested under the first one, which didn't make sense to me. So the only diff with the bottom three is indentation. I added the test on L331.

File: test/metabase/api/search_test.clj [hunk 2]
```diff
@@ -327,26 +327,31 @@
...
-    (testing "you should not be able to see a Table if the current user doesn't have permissions for that Table"
-      (mt/with-temp* [Database [{db-id :id}]
-                      Table    [table {:db_id db-id}]]
-        (perms/revoke-permissions! (group/all-users) db-id)
-        (is (= []
-               (binding [*search-request-results-database-id* db-id]
-                 (search-request :rasta :q (:name table)))))))))
+               (search-request user :q "Round Table"))))))
+  (testing "You should be able to search by their display name"
+    (mt/with-temp Table [table {:name "Round Table" :display_name "Lancelot's Favorite Furniture"}]
```
Comment (camsaul) [inline test/metabase/api/search_test.clj:332]: 😆

**Review by camsaul — APPROVED** [review]
looks good to me

**Comment by mazameli** [comment]
This works correctly and is a welcome improvement. 

But this feels weird because, as a separate issue, we're showing the _underlying_ table name on the search results, where everywhere else we show the display name. That makes for some confusing UX. See in this example, I've renamed a table called `Batting` to `Batter Seasons`, so I can now correctly search by the term "seasons" and find the table I'm looking for — but on the search results page the table is called `Batting`, making me wonder if I've found the right table:

table-names

**Comment by tsmacdonald** [comment]
@mazameli That can be fixed with this change:

```
$ git diff
diff --git a/frontend/src/metabase/lib/entities.js b/frontend/src/metabase/lib/entities.js
index 9b80197f8..65ec0d0a6 100644
--- a/frontend/src/metabase/lib/entities.js
+++ b/frontend/src/metabase/lib/entities.js
@@ -549,7 +549,7 @@ export function createEntity(def: EntityDefinition): Entity {
 
   entity.objectSelectors = {
     getName(object) {
-      return object.name;
+      return object.display_name || object.name;
     },
```

but I'm _very_ afraid of unintended consequences there.

For now, I'll merge this PR into the hub branch and file an issue for the table display so we don't lose track of it. We may need some other display changes too as we expand the fields we search in, add "verified" badges, etc.

**Comment by mazameli** [comment]
Sounds good.

**Pull request merged**